demisto_sdk/dev_tools/spell_checker.py

```diff
@@ -1,95 +1,129 @@
         if self.path.endswith('.md'):
             with open(self.path, 'r') as md_file:
-                md_data = md_file.readlines()
+                md_file_lines = md_file.readlines()
+
+            self.check_md_file(md_file_lines=md_file_lines)
 
-            self.check_md_file(md_data=md_data, unknown_words=unknown_words)
         elif self.path.endswith('.yml'):
             with open(self.path, 'r') as yaml_file:
                 yml_info = yaml.safe_load(yaml_file)
 
-            self.check_yaml(yml_info=yml_info, unknown_words=unknown_words)
+            self.check_yaml(yml_info=yml_info)
 
         else:
             print_error("The file {} is not supported for spell-check command.\n"
                         "Only .yml or .md files are supported.".format(self.path))
-            return 1
+            return False
 
-        if unknown_words:
-            print_error(u"Found the problematic words in {}:\n{}".format(self.path, '\n'.join(unknown_words)))
-            return 1
+        if self.unknown_words:
+            print_error(u"Words that might be misspelled were found in {}:\n\n{}".format(
+                self.path, '\n'.join(self.unknown_words)))
+            return False
 
-        print_color("No problematic words found in {}".format(self.path), LOG_COLORS.GREEN)
-        return 0
+        print_color("No misspelled words found in {}".format(self.path), LOG_COLORS.GREEN)
+        return True
 
-    def check_md_file(self, md_data, unknown_words):
+    def check_md_file(self, md_file_lines):
         """Runs spell check on .md file. Adds unknown words to given unknown_words set.
 
         Args:
-            md_data (list): A line list of the .md file contents
-            unknown_words (set): A set of found unknown words
-
-        Returns:
-            None.
+            md_file_lines (list): A line list of the .md file contents
         """
-        for line in md_data:
+        for line in md_file_lines:
             for word in line.split():
                 if word.isalpha() and self.spellchecker.unknown([word]):
-                    unknown_words.add(word)
+                    self.unknown_words.add(word)
 
-    def check_yaml(self, yml_info, unknown_words):
+    def check_yaml(self, yml_info):
         """Runs spell check on .yml file. Adds unknown words to given unknown_words set.
 
         Args:
             yml_info (Loader): A line list of the .md file contents
-            unknown_words (set): A set of found unknown words
-
-        Returns:
-            None.
         """
+        # yml file is parsed as a large dictionary. Each key in the large dictionary
+        # has a value which can be comprised of a string, a sub-dictionary or a list of dictionaries.
+        # The following code separates the spell check to these cases.
         for key, value in yml_info.items():
-            if key in DISPLAYABLE_LINES:
+            # case 1: a String.
```

Comment:
Let's change to case 1:string value of a user visible field or something like that

Reply:
done